frontend/app/components/thread/thread.tsx

```diff
@@ -29,10 +21,21 @@ interface OwnProps {
   getPreview(text: string): Promise<string>;
 }
 
-type Props = OwnProps & ReturnType<typeof mapStateToProps>;
+const commentSelector = (id: string) => (state: StoreState) => {
+  const { theme, comments, childComments } = state;
+  const comment = comments[id];
+  const childs = childComments[id];
+  const collapsed = getThreadIsCollapsed(comment)(state);
+
+  return { comment, childs, collapsed, theme };
+};
 
-function Thread(props: RenderableProps<Props>) {
-  const { collapsed, comment, childs, level, theme } = props;
+export const Thread: FunctionComponent<Props> = ({ id, level, mix, getPreview }) => {
+  const dispatch = useDispatch();
+  const { collapsed, comment, childs, theme } = useSelector(commentSelector(id));
```

Comment:
Please use shallow equality https://react-redux.js.org/next/api/hooks#equality-comparisons-and-updates
 or use separate useSelector